dotnet/src/Extensions/PromptTemplate.Handlebars/HandlebarsPromptTemplate.cs

```diff
@@ -1,60 +1,72 @@
+/// </summary>
+public class HandlebarsPromptTemplate : IPromptTemplate
 {
     /// <summary>
-    /// Constructor for PromptTemplate.
+    /// Default options for built-in Handlebars helpers.
+    /// </summary>
+    /// TODO [@teresaqhoang]: Support override of default options
+    private readonly HandlebarsPromptTemplateOptions _options = new();
+
+    /// <summary>
+    /// Constructor for Handlebars PromptTemplate.
     /// </summary>
     /// <param name="promptConfig">Prompt template configuration</param>
     /// <param name="loggerFactory">Logger factory</param>
-    public HandlebarsPromptTemplate(PromptTemplateConfig promptConfig, ILoggerFactory? loggerFactory = null)
+    /// <param name="options">Handlebars promnpt template options</param>
+    public HandlebarsPromptTemplate(PromptTemplateConfig promptConfig, ILoggerFactory? loggerFactory = null, HandlebarsPromptTemplateOptions? options = null)
     {
         this._loggerFactory = loggerFactory ?? NullLoggerFactory.Instance;
         this._logger = this._loggerFactory.CreateLogger(typeof(HandlebarsPromptTemplate));
         this._promptModel = promptConfig;
+        this._options = options ?? new();
     }
 
     /// <inheritdoc/>
-#pragma warning disable CS1998 // Async method lacks 'await' operators and will run synchronously
     public async Task<string> RenderAsync(Kernel kernel, KernelArguments? arguments = null, CancellationToken cancellationToken = default)
-#pragma warning restore CS1998
     {
-        var handlebars = HandlebarsDotNet.Handlebars.Create();
+        arguments = this.GetVariables(arguments);
+        var handlebarsInstance = HandlebarsDotNet.Handlebars.Create();
 
-        foreach (IKernelPlugin plugin in kernel.Plugins)
+        // Add helpers for kernel functions
+        KernelFunctionHelpers.Register(handlebarsInstance, kernel, arguments, this._options.PrefixSeparator, cancellationToken);
+
+        // Add built-in system helpers
+        KernelSystemHelpers.Register(handlebarsInstance, arguments, this._options);
+
+        // Register any custom helpers
+        if (this._options.RegisterCustomHelpers is not null)
         {
-            foreach (KernelFunction function in plugin)
-            {
-                handlebars.RegisterHelper($"{plugin.Name}_{function.Name}", (writer, hcontext, parameters) =>
-                {
-                    var result = function.InvokeAsync(kernel, arguments).GetAwaiter().GetResult();
-                    writer.WriteSafeString(result.ToString());
-                });
-            }
+            this._options.RegisterCustomHelpers(handlebarsInstance, arguments);
         }
 
-        var template = handlebars.Compile(this._promptModel.Template);
+        var template = handlebarsInstance.Compile(this._promptModel.Template);
+
+        var prompt = template(arguments);
 
-        return template(this.GetVariables(arguments));
+        return await Task.FromResult(prompt).ConfigureAwait(true);
```

Comment:
As a rule of thumb, you want this to be ConfigureAwait(false) in libraries, including here.